File: Na prekonvertovanie/Fuzzification Tool/Test/FuzzificationHierarchicalEntropy.cs
using System;
using Datasets;

namespace FuzzificationLibrary
{
    public class FuzzificationHierarchicalEntropy : FuzzificationEntropy
    {
        public const double Theta = 0.01;
        public FuzzificationHierarchicalEntropy(DataSets dataToTransform) : base(dataToTransform){}
        public double Treshold { get; set; }
        public override void LastStepInFuzzification(int dimension, int interval)
        {var MeanFE = TotalEntropy[dimension][interval]/DataToTransform.DatasetSize;}

        protected override bool ConditionForStopingFuzzificationInDimension(int dimension, double totalEntropyI,
            double totalEntropyIPrevious)
        {
            Treshold = -Math.Log((double) 1/DataToTransform.OutputIntervals)*(Intervals[dimension] - 1)*Theta;
            Console.WriteLine("Treshold = " + Treshold);
            return (totalEntropyI > totalEntropyIPrevious) || (Math.Abs(totalEntropyI) < 0.00000001) ||
                   (totalEntropyI < Treshold);
        }
    }
}

File: zdrojove kody v c++/FuzzificationHierarchicalEntropy.cpp
#include "FuzzificationHierarchicalEntropy.h"
#include "DataSets.h"

using namespace Datasets;

namespace FuzzificationLibrary
{
	FuzzificationHierarchicalEntropy::FuzzificationHierarchicalEntropy(DataSets* dataToTransform) : FuzzificationEntropy(dataToTransform)
	{
	}

	double FuzzificationHierarchicalEntropy::getTreshold() const
	{
		return _treshold;
	}

	void FuzzificationHierarchicalEntropy::setTreshold(const double& value)
	{
		_treshold = value;
	}

	void FuzzificationHierarchicalEntropy::LastStepInFuzzification(int dimension, int interval)
	{
		auto MeanFE = getTotalEntropy()[dimension][interval] / getDataToTransform()->getDatasetSize();
	}

	bool FuzzificationHierarchicalEntropy::ConditionForStopingFuzzificationInDimension(int dimension, double totalEntropyI, double totalEntropyIPrevious)
	{
		setTreshold(-std::log(static_cast<double>(1) / getDataToTransform()->getOutputIntervals()) * (getIntervals()[dimension] - 1) * Theta);
		std::wcout << L"Treshold = " << getTreshold() << std::endl;
		return (totalEntropyI > totalEntropyIPrevious) || (std::abs(totalEntropyI) < 0.00000001) || (totalEntropyI < getTreshold());
	}
}
